### PR TITLE
capdo: add periodic job to clean DO account

### PR DESCRIPTION
**Pull request by cpanato**
need this first: https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/213

/hold

/assign @prksu @xmudrii @MorrisLaw

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml [hunk 1]
```diff
@@ -1,4 +1,34 @@
 periodics:
+- name: periodic-cluster-api-provider-digitalocean-janitor
+  decorate: true
+  decoration_config:
+    timeout: 1h
+  interval: 12h
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
```
Comment (xmudrii) [inline config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml:9]: Tiny nit: as far as I understand, we run janitor as a (Go) binary. Do we need those two presets in that case?

File: config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+      base_ref: master
+      path_alias: "sigs.k8s.io/cluster-api-provider-digitalocean"
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20201203-4778e22-master
+      command:
+        - "runner.sh"
+        - "./scripts/ci-janitor.sh"
+      securityContext:
+        privileged: true
```
Comment (xmudrii) [inline config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml:23]: Related to the comment above, do we need privileged securityContext in this case?

**Review by xmudrii — COMMENTED** [review]
I've left two tiny nits but otherwise looks good to me.

File: config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml (hunk 1 above)
Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-periodics.yaml]: that is true, thanks for the catch!
@xmudrii PTAL

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by cpanato** [comment]
PR merged good to go
/hold cancel

**Pull request merged**